pkg/download/list.go

```diff
@@ -33,3 +34,27 @@ func ListHandler(dp Protocol, lggr log.Entry, eng *render.Engine) buffalo.Handle
 		return c.Render(http.StatusOK, eng.String(strings.Join(versions, "\n")))
 	}
 }
+
+// ListHandlerBasic implements GET baseURL/module/@v/list as a basic http.HandlerFunc
+// wrapping it as a buffalo handler
+func ListHandlerBasic(dp Protocol, lggr log.Entry, eng *render.Engine) buffalo.Handler {
```

Comment:
one more thing, let's make this return an http.HandlerFunc, and that the caller will have to wrap. This way, we achieve two things: 

1. when we do the switch we don't have to remove many wrappers, we'll just remove the one function (caller) that does all the wrapping: which is `RegisterHandlers`
2. ListHandlerBasic can be used directly by `RegisterHandlersV2` and so at the end, we can just use this one directly (renaming it to regular `RegisterHandlers`

Reply:
Yeah, looking at this again today I think the wrapping is kind of messy. The issue is that the middleware we're using expects a `ProtocolHandler` which returns a `buffalo.Handler`. I was thinking we could define a `ProtocolHandlerV2` and update a couple of the middleware functions to deal with that. 

Also, turns out you can call `app.Mount` to register a basic `http.Handler` on the buffalo app without having to wrap anything. That might be a cleaner approach.

Reply:
Updated the PR with the V2 approach.